src/init/chainstate.cpp

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <init/chainstate.h>
+
+#include <chainparams.h> // for CChainParams
+#include <rpc/blockchain.h> // for RPCNotifyBlockChange
+#include <util/time.h> // for GetTimeMillis
+#include <util/translation.h> // for bilingual_str
+#include <node/blockstorage.h> // for CleanupBlockRevFiles, fHavePruned, fReindex
+#include <node/context.h> // for NodeContext
+#include <node/ui_interface.h> // for InitError, CClientUIInterface member access
+#include <shutdown.h> // for ShutdownRequested
+#include <timedata.h> // for GetAdjustedTime
+#include <validation.h> // for a lot of things
+
+bool LoadChainstateSequence(bool& fLoaded,
+                            bilingual_str& strLoadError,
+                            bool fReset,
+                            CClientUIInterface& uiInterface,
```

Comment:
re: https://github.com/bitcoin/bitcoin/pull/23280#discussion_r743923908

> In commit "init: Extract chainstate loading sequence"
> 
> Could you please use a new variable here rather than shadowing the global? I was very confused about what changed that required `AddReadErrCallback([](...))` -> `AddReadErrCallback([&](...))`

IMO ideal thing to do here so this commit remains moveonly and easy to review and doesn't require thinking about renames or new variables or adding `[&]`, is just to keep `uiInterface` a global instead of a parameter here, since it is already going away later anyway in 83ed7db27b4c4798666b77d83fc8e7c3eab8c621.

Reply:
Fixed as of 6736e24a490d756e0e1e70b9e212775cf305ebe6